File: python/01-circles-demo/py/step-04.py
# Demo 1: Animate the Circles (Processing.py)
# Step 4: Animated grid of circles
# For use in Processing IDE, Python mode.

cols = 10
rows = 10
diameter = 40
padding = 20

def setup():
    size(600, 600)
    noStroke()
    frameRate(30)

# Animate the circles in draw()
def draw():
    background(255)
    for y in range(rows):
        for x in range(cols):
            cx = padding + x * (diameter + padding)
            cy = padding + y * (diameter + padding)
            # Animate diameter with sin() for a pulsing effect
            t = frameCount * 0.1 + x + y
            d = diameter + 20 * sin(t)
            fill(80, 180, 255)
            ellipse(cx, cy, d, d)

File: javascript/01-circles-demo/js/step-04.js
// Save output to your computer (usually Downloads folder)
function keyPressed() {
  if (key === 's' || key === 'S') {
    // The browser will save the file to your default downloads folder.
    saveCanvas('animated-circles', 'jpg'); // Save canvas as image
  }
}

// Step 4: Grid of circles with random radius and color
// This sketch uses random() to vary the size and color of each circle

function setup() {
  createCanvas(800, 800); // Create a canvas 800x800 pixels
  background(220); // Light gray background
  noStroke(); // No outline on circles
  let numCols = 8; // Number of columns
  let numRows = 8; // Number of rows
  let spacingX = width / (numCols + 1); // Horizontal spacing
  let spacingY = height / (numRows + 1); // Vertical spacing
  for (let i = 1; i <= numCols; i++) {
    for (let j = 1; j <= numRows; j++) {
      let d = random(30, 100); // Random diameter for each circle
      let r = random(255); // Random red value
      let g = random(255); // Random green value
      let b = random(255); // Random blue value
      fill(r, g, b); // Set fill color
      ellipse(i * spacingX, j * spacingY, d); // Draw circle
    }
  }
}

// No animation in this step
